Karma-Khameleons/bangazon-website
# Sam product quantity

## Description
This updates the info presented on the product detail page and allows customers to select product quantity when buying an item. It does this by specifying a "through" join table called LineItem on the CustomerOrder "line_items" Many2Many field. The join table keeps track of product quantity for each line item.



## Related Ticket(s)
Fixed: #49, #54 


## Problem to Solve
Customers should be able to purchase multiple units of a given item

## Changes
Changed the CustomerOrder model to add a through table on the "line_items" M2M field. Added a couple lines to admin.py and models/init.py. Also changed product_detail template to include a quantity option that is tied to the Product's available quantity.


## Expected Behavior
Line items for a CustomerOrder should include the quantity being purchased. Currently no way to subtract from LineItem quantity after it's added to the order. However, the quantity can be increased by adding more from the Product Detail page.


## Steps to Test Solution
1. From the root of the Django project, run the following commands:
```
git fetch --all
git checkout origin/sam-ProductQuantity
rm -rf bang_app/migrations
python manage.py makemigrations bang_app
python manage.py migrate
python manage.py runserver
```
2. As a registered user add items to your order
3. From the ```/admin/bang_app/lineitem/``` page, ensure that the most recently added LineItem matches the one you just created in all aspects, including quantity.


## Unit Testing

## Documentation


## Deploy Notes
As shown above in the "Steps to Test" section, new migrations will need to be made. In addition, an entire recreation of the database COULD be necessary. I honestly don't remember if I had to do that. Will update this section if it becomes an issue.
